jupyter_core/tests/test_application.py

```diff
@@ -27,3 +29,12 @@ def test_custom_config():
         app.initialize(['--config', f.name])
     assert app.config_file == f.name
     assert app.n == 10
+
+
+def test_cli_override():
+    app = DummyApp()
+    with NamedTemporaryFile(suffix='.py', mode='w') as f:
```

Comment:
Will this work on Windows? It needs to open two handles on the file at the same time, and I seem to remember that's problematic. We have `NamedFileInTemporaryDirectory` in genutils for this.